### PR TITLE
Argument checking

### PR DESCRIPTION
**Pull request by treilly94**

### CODE REVIEW

File: tests/test_commandline.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+            valid_write_option("pandas")
+        err = str(cm.exception)
+        self.assertEqual("pandas isn't a valid write option. Must be either w or a", err)
+
+        # Assert doesn't raise error when valid
+        for i in ["w", "a"]:
+            output = valid_write_option(i)
+            self.assertEqual(i, output)
+
+    def test_get_args_wrap(self):
```
Comment (KayleighB94) [inline tests/test_commandline.py]: what happens if one of them is missing

File: tests/test_commandline.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        args_in = ["wrap", "-m", "./tests/__init__.py", "-s", "./tests/test_commandline.py"]
+        args_out = get_args(args_in)
+        self.assertEqual("wrap", args_out.command)
+        self.assertEqual("./tests/__init__.py", args_out.method_config)
+        self.assertEqual("./tests/test_commandline.py", args_out.structure_config)
+
+    def test_get_args_parse(self):
```
Comment (KayleighB94) [inline tests/test_commandline.py]: What happens when they aren't passed in

File: wrapper_writer/commandline.py [hunk 3]
```diff
@@ -1,59 +1,136 @@
...
+    parser_parse.add_argument('-d', '--directory', type=existing_directory, default=None,
+                              help='The absolute path to a directory containing files to be parsed')
+    parser_parse.add_argument('-t', '--target-format', default=".*\.scala",
+                              help='The format of the files to read from the directory')
+
+    # Parse the argument lists
+    return parser.parse_args(args)
+
+
+def parse(args):
```
Comment (KayleighB94) [inline wrapper_writer/commandline.py]: Possible a test for this, to check it calls the right functions

File: wrapper_writer/commandline.py [hunk 4]
```diff
@@ -1,59 +1,136 @@
...
+        p.files.extend(args.files.split(","))
+    if args.directory:
+        p.get_files(args.directory, args.target_format)
+
+    p.multi_process()
+
+    p.write_config()
+
+
+def wrap(args):
```
Comment (KayleighB94) [inline wrapper_writer/commandline.py]: Possible a test for this, to check it calls the right functions

/ do we need a function for this one line

File: tests/test_commandline.py (hunk 1 above)
Comment (treilly94) [reply tests/test_commandline.py]: It throws an error

File: tests/test_commandline.py (hunk 2 above)
Comment (treilly94) [reply tests/test_commandline.py]: They default

File: wrapper_writer/commandline.py (hunk 4 above)
Comment (treilly94) [reply wrapper_writer/commandline.py]: Not really. I originally split them out to add some error handling around them, but thats all done within argparse now.
Might still make them easier to test though and allow us to add functionality in the future

**Pull request merged**